src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts

```diff
@@ -158,22 +234,52 @@ class encryptionSettings {
                 }
             }]
         });
+        
+        this.enabled.subscribe(enabled => {
+            if (enabled && this.enabled.isValid()) {
+                this.mode("UseProvidedKey");
+                
+                if (!this.originalKey()) {
+                    return this.encryptionSection.generateEncryptionKey();
+                }
+            }
+            
+            if (!this.enabled()) {
+                this.changeKeyRequest(false);
+            }
+        });
 
-        const keyConfirmationNeeded = ko.pureComputed(() => this.canProvideOwnKey() && this.mode() === "UseProvidedKey");
+        this.changeKeyRequest.subscribe(changeKeyRequest => {
+            if (changeKeyRequest) {
+                return this.encryptionSection.generateEncryptionKey();
+            }
+        });
         
+        const keyConfirmationNeeded = ko.pureComputed(() => (this.canProvideOwnKey() && !this.originalKey()) ||
```

Comment:
can we simply this expression?

Reply:
changed to:
```
        const keyConfirmationNeeded = ko.pureComputed(() =>
             this.canProvideOwnKey() &&
             (!this.originalKey() || this.originalKey() && this.changeKeyRequest())
        );
```

Reply:
could you put `(` and `)` around. Please never use `||` and `&&`  together without braces.